lib/Rx/Operator/RetryWhenOperator.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Rx\Operator;
+
+use Rx\Disposable\CompositeDisposable;
+use Rx\Disposable\EmptyDisposable;
+use Rx\ObservableInterface;
+use Rx\Observer\CallbackObserver;
+use Rx\ObserverInterface;
+use Rx\SchedulerInterface;
+use Rx\Subject\Subject;
+
+class RetryWhenOperator implements OperatorInterface
+{
+    private $notificationHandler;
+
+    /**
+     * RetryWhenOperator constructor.
+     *
+     * @param callable $notificationHandler
+     */
+    public function __construct(callable $notificationHandler)
+    {
+        $this->notificationHandler = $notificationHandler;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function __invoke(
+        ObservableInterface $observable,
+        ObserverInterface $observer,
+        SchedulerInterface $scheduler = null
+    ) {
+        $errors           = new Subject();
+        $when             = call_user_func($this->notificationHandler, $errors->asObservable());
+        $disposable       = new CompositeDisposable();
+        $sourceDisposable = new EmptyDisposable();
+        $innerCompleted   = false;
```

Comment:
We shouldn't keep state in the operators themselves I think. Nothing is stopping you from doing:

```
$myOperator = new MyOperator();
$o1->lift($myOperator)->subscribe(/* ... */);
$o2->lift($myOperator)->subscribe(/* ... */);
```

That would get messy?